src/axom/quest/detail/DistributedClosestPointImpl.hpp

```diff
@@ -0,0 +1,1350 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_DISTRIBUTED_CLOSEST_POINT_IMPL_H_
+#define QUEST_DISTRIBUTED_CLOSEST_POINT_IMPL_H_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/spin.hpp"
+#include "axom/core/execution/runtime_policy.hpp"
+
+#include "axom/fmt.hpp"
+
+#include "conduit_blueprint.hpp"
+#include "conduit_blueprint_mcarray.hpp"
+#include "conduit_blueprint_mpi.hpp"
+#include "conduit_relay_mpi.hpp"
+#include "conduit_relay_io.hpp"
+
+#include <memory>
+#include <limits>
+#include <cstdlib>
+#include <cmath>
+#include <vector>
+
+#ifndef AXOM_USE_MPI
+  #error This file requires Axom to be configured with MPI
+#endif
+#include "mpi.h"
+
+namespace axom
+{
+namespace quest
+{
+namespace internal
```

Comment:
I'm confused. You've moved all of these methods from one header file to another. Most of these will not be inlined and, if included, in multiple header files may be parsed multiple times. Would it be better to move these to a source (*.cpp) file?

Reply:
@rhornung67 That's a good point with respect to the `DistributedClosestPoint` class.  Its methods can be moved a `DistributedClosestPoint.cpp` file.  The templated `DistributedClosestPointImpl` class will then be parsed only once, when included in the `.cpp` file.  Let me know if that doesn't address your point.